llvm/include/llvm/Analysis/DXILMetadataAnalysis.h

```diff
@@ -19,12 +21,20 @@ namespace llvm {
 
 namespace dxil {
 
+struct EntryProperties {
+  // Specific target shader stage may be specified for entry functions
+  Triple::EnvironmentType ShaderStage = Triple::UnknownEnvironment;
+  unsigned NumThreadsX; // X component
+  unsigned NumThreadsY; // Y component
+  unsigned NumThreadsZ; // Z component
+};
+
 struct ModuleMetadataInfo {
   VersionTuple DXILVersion{};
   VersionTuple ShaderModelVersion{};
   Triple::EnvironmentType ShaderStage = Triple::UnknownEnvironment;
   VersionTuple ValidatorVersion{};
-
+  MapVector<Function *, EntryProperties> EntryPropertyMap;
```

Comment:
Why a MapVector? Is the insertion order of these functions interesting to preserve for some reason, or is this just here so that it's easy to make the order that we print them in deterministic?